### ISSUE TITLE
Implement proposals service in js-uprcl-server

### ISSUE THREAD

**Issue by pepoospina** [uprtcl/js-uprtcl-server#16]
make [this interface](https://github.com/uprtcl/js-uprtcl/blob/evees.orbit-db.pepo/modules/evees/src/services/proposals.provider.ts#L34) work with with the HTTP server

## In the frontend

- [x] create `evees.proposals.http.ts` under `evees/src/services/providers/http` and create a class `ProposalsHttp` inside (similar to [ProposalsEthereum](https://github.com/uprtcl/js-uprtcl/blob/develop-web2/modules/evees/src/services/providers/ethereum/proposals.ethereum.ts)). 
- [x] Implement all the required methods as calls to the HTTP API (use the HttpConnection class and receive a `host` string, similar to the evees-access-control-https.ts constructor)
- [x] updated the `evees.http.ts` to use the newly created `ProposalsHttp` as its `proposals` property (similar to how it uses the evees-access-control-http.ts).

## In the backend

- [x] create a new folder under services called `proposals`. And add four files: `proposals.controller.ts`, `proposals.services.ts`, `proposals.schema.ts` and `proposals.repository.ts` (use the known ources folder as reference, it's a simple one)
- [x] define dgraph schema under proposals.schema.ts. It should be something like [this example](https://gist.github.com/pepoospina/dfaff7216d9749fa8448d514c3c79785).
- [x] define dgraph repository. It should create a proposal ([use this as an example](https://gist.github.com/pepoospina/940035537032ffd18fe02994cbf10e02))
- [x] then create the proposals.service and the proposals.controller.

**Comment by sotous** [comment]
Regarding the interface mentioned in the description, about this method:

`  createAndPropose(
    newPerspectivesData: NewPerspectiveData[],
    proposal: NewProposal
  ): Promise<string>;`

How are we suppose to provide the following information: 

`export interface NewProposal {
  fromPerspectiveId: string;
  toPerspectiveId: string;
  fromHeadId: string;
  toHeadId: string;
  updates: UpdateRequest[];
}`

If we are creating perspectives from nothing?, I think that the frontend should only provide the newPerspectivesData, following the creation of the proposal in the server.

**Comment by pepoospina** [comment]
### Proposals Logic

A proposal is created by a user. It has `toPerspective` and `fromPerspective`, and includes a number of `updates`. The proposal is in state OPEN and the logged user is used as the creator (not the `creatorId` in the proposal property, because that one can be counterfeited).

Once OPEN, the creator of a proposal can DECLINE it, in which case its state changes to DECLINED. 

Any user who has canEdit access too all the perspectives in the `updates` array (not in toPerspective), can set the state to REJECTED.

Any user that has `canEdit` permissions to all the perspectives in the `updates` (not in toPerspective), can authorize the proposal. Once authorized, the state passes to EXECUTED.

When authorized, the `ProposalService` uses the `updatePerspective()` method from `UprtclService`, and send it the `loggedUser` derived from the current request. This means the logged user that calls `authorize()` **must** have `canEdit` permissions (or `canAdmin`) on all the perspectives in the `updates` array of the proposal. 

The `getPerspectiveProposals()` function should remove the CANCELLED and DECLINED proposals from its list and include only OPEN and EXECUTED (this can be done in the frontend for now).

**Comment by sotous** [comment]
OK guys, seems like the backend tasks are done. Here you can take a look at the latest commits:

-  https://github.com/uprtcl/js-uprtcl-server/compare/develop...sotous:feature/proposals-service

We still need to discuss some refactoring topics, such as:

1. The enumeration in `dgraph` schema. In which I made some research.
2. Move all proposal `types` to its own `type` file.
3. Proposal rejection possibilities with empty updates on that proposal.

I'll jump to the front-end API integration now.

**Comment by sotous** [comment]
Hello!

I have some news. I got to handle all the functionality in the front-end and in the back-end. However, we still need to do some refactor tasks, including the ones mentioned in the previous comment:

- [ ] To move the `canAuthorize` method out of the `proposalsRepository`. Repositories should only contain functions that interact with the DB.

Furthermore, I got inconsistency issues doing UI and manual tests that we have to discuss ASAP.

You guys can check out the latest modifications for the front-end here:

- https://github.com/uprtcl/js-uprtcl/compare/develop-web2...sotous:feature/proposals-service

And for the server, here:

- https://github.com/uprtcl/js-uprtcl-server/compare/develop...sotous:feature/proposals-service
